src/Symfony/Component/Messenger/Middleware/LockMiddleware.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware;
+
+use Symfony\Component\Lock\LockFactory;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Message\LockableMessage;
+use Symfony\Component\Messenger\Stamp\LockStamp;
+use Symfony\Component\Messenger\Stamp\ReceivedStamp;
+
+final class LockMiddleware implements MiddlewareInterface
+{
+    private ?LockFactory $lockFactory = null;
+
+    public function __construct(
+        ?LockFactory $lockFactory,
+    ) {
+        $this->lockFactory = $lockFactory;
+    }
+
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+        if (null === $this->lockFactory) {
+            return $stack->next()->handle($envelope, $stack);
+        }
+
+        $message = $envelope->getMessage();
+
+        // If we're trying to dispatch a lockable message.
+        if ($message instanceof LockableMessage && null === $envelope->last(ReceivedStamp::class)) {
+            $key = $message->getKey();
+
+            if (null !== $key) {
+                // The acquire call must be done before stamping the message
+                // in order to have the full state of the key in the stamp.
+                $canAcquire = $this->lockFactory->createLockFromKey($key, autoRelease: false)->acquire();
```

Comment:
IMHO LockTTL should be configurable

Reply:
Should it be another interface to implement ?
-> If you implements LockMessage => you have the default TTL
-> If you implements something like ConfigurableLockMessage => You have to return a custom TTL 

Or should both method be in the LockMessage interface ?

Reply:
I don't have strong opinion on this (the same question for the other comment about `release before handle` thing)

Reply:
I think:
- 3 interfaces would be too much
- TTL might not be often configured

So I tried with two interfaces:
- LockableMessageInterface
- TTLAwareLockableMessageInterface